aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/GetObjectRequest.java

```diff
@@ -1008,4 +1008,47 @@ public GetObjectRequest withS3ObjectId(S3ObjectId s3ObjectId) {
         setS3ObjectId(s3ObjectId);
         return this;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) return true;
+        if (object == null || getClass() != object.getClass()) return false;
```

Comment:
Thanks - sorry to be a stickler but when faced with these sorts of decisions I usually opt for consistency - it's important that we do our best to ensure the SDK behaves in a predictable way.